Source/playerdat.cpp

```diff
@@ -152,11 +153,94 @@ void ReloadExperienceData()
 	}
 }
 
+void LoadClassAttributes(std::string_view classPath, ClassAttributes &out)
+{
+	const std::string filename = StrCat("txtdata\\classes\\", classPath, "\\attributes.tsv");
+	tl::expected<DataFile, DataFile::Error> dataFileResult = DataFile::load(filename);
+	if (!dataFileResult.has_value()) {
+		DataFile::reportFatalError(dataFileResult.error(), filename);
+	}
+	DataFile &dataFile = dataFileResult.value();
+
+	if (tl::expected<void, DataFile::Error> result = dataFile.skipHeader();
+	    !result.has_value()) {
+		DataFile::reportFatalError(result.error(), filename);
+	}
+
+	auto recordIt = dataFile.begin();
+	const auto recordEnd = dataFile.end();
+
+	const auto getValueField = [&](std::string_view expectedKey) {
+		if (recordIt == recordEnd) {
+			app_fatal(fmt::format("Missing field {} in {}", expectedKey, filename));
+		}
+		DataFileRecord record = *recordIt;
+		FieldIterator fieldIt = record.begin();
+		const FieldIterator endField = record.end();
+
+		const std::string_view key = (*fieldIt).value();
+		if (key != expectedKey) {
+			app_fatal(fmt::format("Unexpected field in {}: got {}, expected {}", filename, key, expectedKey));
+		}
+
+		++fieldIt;
+		if (fieldIt == endField) {
+			DataFile::reportFatalError(DataFile::Error::NotEnoughColumns, filename);

```

Comment:
Oh, turns out in the WIP branch I hadn't pushed yet I do exactly this, create a DataFileField::Error enum and use it instead of std::errc in reportFatalFieldError

Reply:
Also just realised I never cleaned up the use of this when parsing experience data... no need to worry about it in this one since there's already convention.